### PR TITLE
Intro text for General Concepts and Building Extensions

### PR DESCRIPTION
**Pull request by robbiejackson**

### CODE REVIEW

**Comment by HLeithner** [comment]
Looks good.

**Comment by Fedik** [comment]
If it is possible. I suggest to add some hint for developers, what extension type to use and when.

For newby it is often hard to decide what they need: component, module, plugin.
Developers who come from other CMS (Drupal or WP, etc) where only one type "plugin", starts coding plugin, and after a while they realise they need a component, or module.

File: versioned_docs/version-4.4/building-extensions/index.md [hunk 1]
```diff
@@ -2,10 +2,51 @@
...
 
-This page is unfinished, please use the **Edit this Page** link at the bottom of this page to help make it more useful.
+![Screenshot showing extension types](./_assets/screenshot-extension-types.jpg)
 
-:::
+# Components
+[Components](./components/index.md) provide the central part of a web page on a Joomla site; each site web page displays the output from one component. They can be thought of as mini applications. Most components have two parts: a site part and an administrator part. For example, `com_content` is the component which handles articles; on the site front-end `com_content` displays articles to website visitors and on the back-end `com_content` provides the functionality for administrators to edit articles.
```
Comment (Fedik) [inline versioned_docs/version-4.4/building-extensions/index.md:11]: For components, I would add their main role. Kind of: Their main role is to manage the Data on the site, provides an interface for data creatin/removing/editting etc. They also process the Request. 

Manage Articles, Categories (Content component), manage products and orders (Shoping cart component), manage images and media files (Gallery and media component).

Extending functionality: like provide backup solutions, handle special request.

Maybe with some better wording :smiley:

File: versioned_docs/version-4.4/building-extensions/index.md [hunk 2]
```diff
@@ -2,10 +2,51 @@
...
+
+- Examples: Content (com_content), Banners (com_banners), Contacts (com_contact), News Feeds (com_newsfeeds)
+- Management feature: Admin menu → Components → Contacts (for com_contact)
+
+# Modules
+[Modules](./modules/index.md) are more lightweight and flexible extensions used within page rendering. These modules are mostly known as the “boxes” that are arranged around a component, for example: the login module or the breadcrumbs module. Modules are assigned per menu item. So you can decide to show or hide the login module depending on which menu item the user is viewing. Sometimes modules are linked to components such as the “Latest News” module which links to the `com_content` and displays links to the newest content items. However, modules do not need to be linked to components, as a matter of fact they don't even need to be linked to anything and can be just static HTML or text.
```
Comment (Fedik) [inline versioned_docs/version-4.4/building-extensions/index.md:17]: >  Sometimes modules are linked to components ...

In general the module is a companion of the component. 
The role is to display a short teaser of component data (latest news), or specific data that come from component (like menu, image slider).

But also true, that module can be standalone, to display some weather info, or stuff like that.

**Comment by Fedik** [comment]
I have added a couple notes, hope it will be helpfull.

Thanks for you work.

**Comment by robbiejackson** [comment]
Thanks for the suggestions @Fedik I've added some text along the lines you suggest.

**Pull request merged**